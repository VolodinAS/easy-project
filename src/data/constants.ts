export const localStorageTokenKey = 'easy-project.auth-token';