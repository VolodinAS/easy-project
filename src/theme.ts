const theme = {
    color: {
        typeTask: '#F8BFBF',
        typeFeature: '#B7CFFF',
        typeBug: '#FEFBBD',
        black: '#000000',
        lightPurple: '#CFC7D9',
        purple: '#7A52AA',
        lightGray: '#F5F5F5',
        darkGray: '#9C9C9C',
        red: '#EB0C0C',
        white: '#FFFFFF',
        smallText: '#999999',
        opacityGray: '#ACACAC',
        landingColors: {
            mainGray: '#D5E6EA',
            mainBlue: '#3848A9',
            benefitsItemBackground: '#EDD8BE',
            buttonStartBackground: '#E43958',
            footerBackground: '#343F53',
            headerLogo: '#010101',
            headerBackground: '#F1E7DA'
        },
        dashboardColors: {
            overlayButtonCreate: '#ABE0E7',
            overlayButtonYes: '#B4E7AB',
            overlayButtonNo: '#F9CCBE',
        }
    }
};

export default theme;
