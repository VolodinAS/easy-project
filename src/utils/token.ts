import { localStorageTokenKey } from '../data/constants';

export const authToken = {
    get token() {
        return localStorage.getItem(localStorageTokenKey);
    },
    set token(token) {
        localStorage.setItem(localStorageTokenKey, token);
    },
    checkToken() {
        return Boolean(localStorage.getItem(localStorageTokenKey));
    },
    clearToken() {
        localStorage.removeItem(localStorageTokenKey);
    }
};