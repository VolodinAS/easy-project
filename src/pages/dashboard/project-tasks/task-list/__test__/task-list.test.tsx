/* eslint-disable check-file/folder-naming-convention */
import React from 'react';
import { describe, it, expect, jest as jestGlobal, beforeAll, afterAll } from '@jest/globals';
import { render, waitFor } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';
import { StaticRouter } from 'react-router-dom/server';
import { setupServer } from 'msw/node';
import { rest } from 'msw';
import { Provider } from 'react-redux';
// import { useNavigate } from 'react-router-dom';
import 'whatwg-fetch';

import theme from '../../../../../theme';
import taskListData from '../../../../../../stubs/json/dashboard/task-list/success.json';
import projectListData from '../../../../../../stubs/json/dashboard/projects-list/success.json';
import DashboardRoutes from '../../../../../dashboard-routes';
import { store } from '../../../../../data/store';
import { authToken } from '../../../../../utils/token';

// jestGlobal.mock('react-router-dom', () => {

//     const originalModule = jest.requireActual('react-router-dom') as object;

//     const navigate = jest.fn();

//     return {
//         __esModule: true,
//         ...originalModule,
//         useNavigate: () => navigate
//     };
// });

const serverSuccess = setupServer(
    rest.get('*/projects', (req, res, ctx) => {
        return res(
            ctx.json(projectListData)
        );
    }),
    rest.get('*/tasks/2', (req, res, ctx) => {
        return res(
            ctx.json(taskListData)
        );
    })
);

describe('Task List', () => {
    beforeAll( () => {
        serverSuccess.listen();
    } );
    afterAll( () => {
        serverSuccess.close();
    } );
    it('render', async () => {

        authToken.token = 'token';

        const { container, getByText } = render(
            <StaticRouter location={'/easy-project/dashboard/2'}>
                <Provider store={store}>
                    <ThemeProvider theme={theme}>
                        <DashboardRoutes/>
                    </ThemeProvider>
                </Provider>
            </StaticRouter>
        );

        

        await waitFor(() => {
            const groupTitle = getByText(/todo \(8\)/i);
            expect(container).toMatchSnapshot();
        }, { timeout: 15000 });

        
    });


});