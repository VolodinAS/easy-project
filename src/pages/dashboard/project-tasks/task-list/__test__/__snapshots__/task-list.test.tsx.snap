// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task List render 1`] = `
<div>
  <div
    class="css-1boxdi0"
  >
    <div
      class="css-aoi44d"
    >
      <div
        class="css-174g2o2"
      >
        <a
          class="css-13lnaem"
          href="/easy-project/dashboard"
          title="Дашборд"
        >
          <div
            class="css-1povu0j"
          >
            <div
              class="css-4ucq2w"
            >
              <img
                alt="Easy project logo"
                src="test-file-stub"
              />
            </div>
            <p
              class="css-1vxbtp5"
            >
              <span>
                easy
              </span>
              <span>
                project
              </span>
            </p>
          </div>
        </a>
        <nav
          class="css-83k1q7"
        >
          <div
            class="css-16uff1u"
          />
          <div
            class="css-16uff1u"
          >
            <button
              class="css-1640pth"
              title="language select"
              value="ru"
            >
              <img
                class="css-zlc08d"
                src="test-file-stub"
              />
            </button>
            <div
              class="css-746kv5"
            />
            <a
              class="css-13xx20t"
              href="/easy-project/dashboard"
              title="Выход"
            >
              Выход
            </a>
          </div>
        </nav>
      </div>
    </div>
    <div
      class="css-14f5air"
    >
      <main
        class="css-mx63p3"
      >
        <div
          class="css-44i7do"
        >
          <div
            class="css-3l4y0u"
          />
          <h1
            class="css-1s264hv"
          >
            My second project [можно нажимать]
          </h1>
        </div>
        <div
          class="css-1vbmg59"
        >
          <div
            class="css-1ni1msf"
          >
            <div>
              <p
                class="css-1xdtxed"
              >
                TODO (8)
              </p>
              <a
                class="css-1u1ubl7"
                defaultcolor="#93E586"
                href="/easy-project/dashboard/2/1"
              >
                <div
                  class="css-6sdg1k"
                />
                <div
                  class="css-1lojfjy"
                >
                  <p
                    class="css-1l5tvrx"
                  >
                    Задача с описанием
                  </p>
                </div>
                <div
                  class="css-1ox63rq"
                >
                  <div
                    class="css-1c3mbim"
                  >
                    <div
                      class="css-1e2fy0l"
                    >
                      <img
                        class="css-6jm447"
                        src="test-file-stub"
                      />
                    </div>
                    <div
                      class="css-l273ov"
                    >
                      <p
                        class="css-iz8f7z"
                      >
                        MSP
                        -
                        1
                      </p>
                    </div>
                  </div>
                  <div
                    class="css-1e2fy0l"
                  >
                    <div
                      class="css-11um8xy"
                    />
                  </div>
                </div>
              </a>
              <a
                class="css-1u1ubl7"
                defaultcolor="#93E586"
                href="/easy-project/dashboard/2/2"
              >
                <div
                  class="css-6sdg1k"
                />
                <div
                  class="css-1lojfjy"
                >
                  <p
                    class="css-1l5tvrx"
                  >
                    Добавить первую задачу
                  </p>
                </div>
                <div
                  class="css-1ox63rq"
                >
                  <div
                    class="css-1c3mbim"
                  >
                    <div
                      class="css-1e2fy0l"
                    >
                      <img
                        class="css-6jm447"
                        src="test-file-stub"
                      />
                    </div>
                    <div
                      class="css-l273ov"
                    >
                      <p
                        class="css-iz8f7z"
                      >
                        MSP
                        -
                        1
                      </p>
                    </div>
                  </div>
                  <div
                    class="css-1e2fy0l"
                  >
                    <div
                      class="css-11um8xy"
                    />
                  </div>
                </div>
              </a>
              <a
                class="css-1u1ubl7"
                defaultcolor="#93E586"
                href="/easy-project/dashboard/2/3"
              >
                <div
                  class="css-6sdg1k"
                />
                <div
                  class="css-1lojfjy"
                >
                  <p
                    class="css-1l5tvrx"
                  >
                    Добавить первую задачу
                  </p>
                </div>
                <div
                  class="css-1ox63rq"
                >
                  <div
                    class="css-1c3mbim"
                  >
                    <div
                      class="css-1e2fy0l"
                    >
                      <img
                        class="css-6jm447"
                        src="test-file-stub"
                      />
                    </div>
                    <div
                      class="css-l273ov"
                    >
                      <p
                        class="css-iz8f7z"
                      >
                        MSP
                        -
                        1
                      </p>
                    </div>
                  </div>
                  <div
                    class="css-1e2fy0l"
                  >
                    <div
                      class="css-11um8xy"
                    />
                  </div>
                </div>
              </a>
              <a
                class="css-1u1ubl7"
                defaultcolor="#93E586"
                href="/easy-project/dashboard/2/4"
              >
                <div
                  class="css-6sdg1k"
                />
                <div
                  class="css-1lojfjy"
                >
                  <p
                    class="css-1l5tvrx"
                  >
                    Добавить первую задачу
                  </p>
                </div>
                <div
                  class="css-1ox63rq"
                >
                  <div
                    class="css-1c3mbim"
                  >
                    <div
                      class="css-1e2fy0l"
                    >
                      <img
                        class="css-6jm447"
                        src="test-file-stub"
                      />
                    </div>
                    <div
                      class="css-l273ov"
                    >
                      <p
                        class="css-iz8f7z"
                      >
                        MSP
                        -
                        1
                      </p>
                    </div>
                  </div>
                  <div
                    class="css-1e2fy0l"
                  >
                    <div
                      class="css-11um8xy"
                    />
                  </div>
                </div>
              </a>
              <a
                class="css-1u1ubl7"
                defaultcolor="#93E586"
                href="/easy-project/dashboard/2/5"
              >
                <div
                  class="css-6sdg1k"
                />
                <div
                  class="css-1lojfjy"
                >
                  <p
                    class="css-1l5tvrx"
                  >
                    Добавить первую задачу
                  </p>
                </div>
                <div
                  class="css-1ox63rq"
                >
                  <div
                    class="css-1c3mbim"
                  >
                    <div
                      class="css-1e2fy0l"
                    >
                      <img
                        class="css-6jm447"
                        src="test-file-stub"
                      />
                    </div>
                    <div
                      class="css-l273ov"
                    >
                      <p
                        class="css-iz8f7z"
                      >
                        MSP
                        -
                        1
                      </p>
                    </div>
                  </div>
                  <div
                    class="css-1e2fy0l"
                  >
                    <div
                      class="css-11um8xy"
                    />
                  </div>
                </div>
              </a>
              <a
                class="css-1u1ubl7"
                defaultcolor="#93E586"
                href="/easy-project/dashboard/2/6"
              >
                <div
                  class="css-6sdg1k"
                />
                <div
                  class="css-1lojfjy"
                >
                  <p
                    class="css-1l5tvrx"
                  >
                    Добавить первую задачу
                  </p>
                </div>
                <div
                  class="css-1ox63rq"
                >
                  <div
                    class="css-1c3mbim"
                  >
                    <div
                      class="css-1e2fy0l"
                    >
                      <img
                        class="css-6jm447"
                        src="test-file-stub"
                      />
                    </div>
                    <div
                      class="css-l273ov"
                    >
                      <p
                        class="css-iz8f7z"
                      >
                        MSP
                        -
                        1
                      </p>
                    </div>
                  </div>
                  <div
                    class="css-1e2fy0l"
                  >
                    <div
                      class="css-11um8xy"
                    />
                  </div>
                </div>
              </a>
              <a
                class="css-1u1ubl7"
                defaultcolor="#93E586"
                href="/easy-project/dashboard/2/7"
              >
                <div
                  class="css-6sdg1k"
                />
                <div
                  class="css-1lojfjy"
                >
                  <p
                    class="css-1l5tvrx"
                  >
                    Добавить первую задачу
                  </p>
                </div>
                <div
                  class="css-1ox63rq"
                >
                  <div
                    class="css-1c3mbim"
                  >
                    <div
                      class="css-1e2fy0l"
                    >
                      <img
                        class="css-6jm447"
                        src="test-file-stub"
                      />
                    </div>
                    <div
                      class="css-l273ov"
                    >
                      <p
                        class="css-iz8f7z"
                      >
                        MSP
                        -
                        1
                      </p>
                    </div>
                  </div>
                  <div
                    class="css-1e2fy0l"
                  >
                    <div
                      class="css-11um8xy"
                    />
                  </div>
                </div>
              </a>
              <a
                class="css-1u1ubl7"
                defaultcolor="#60AFF8"
                href="/easy-project/dashboard/2/11"
              >
                <div
                  class="css-3olot6"
                />
                <div
                  class="css-1lojfjy"
                >
                  <p
                    class="css-1l5tvrx"
                  >
                    Создать первый проект
                  </p>
                </div>
                <div
                  class="css-1ox63rq"
                >
                  <div
                    class="css-1c3mbim"
                  >
                    <div
                      class="css-1e2fy0l"
                    >
                      <img
                        class="css-6jm447"
                        src="test-file-stub"
                      />
                    </div>
                    <div
                      class="css-l273ov"
                    >
                      <p
                        class="css-iz8f7z"
                      >
                        MSP
                        -
                        2
                      </p>
                    </div>
                  </div>
                  <div
                    class="css-1e2fy0l"
                  />
                </div>
              </a>
            </div>
            <a
              buttontheme="[object Object]"
              class="css-vkvbx5"
              href="/easy-project/dashboard/2/new-task"
              title="Добавить задачу"
            >
              Добавить задачу
            </a>
          </div>
          <div
            class="css-1ni1msf"
          >
            <div>
              <p
                class="css-1xdtxed"
              >
                INPROGRESS (2)
              </p>
              <a
                class="css-1u1ubl7"
                defaultcolor="#93E586"
                href="/easy-project/dashboard/2/111"
              >
                <div
                  class="css-6sdg1k"
                />
                <div
                  class="css-1lojfjy"
                >
                  <p
                    class="css-1l5tvrx"
                  >
                    Удаление задач
                  </p>
                </div>
                <div
                  class="css-1ox63rq"
                >
                  <div
                    class="css-1c3mbim"
                  >
                    <div
                      class="css-1e2fy0l"
                    >
                      <img
                        class="css-6jm447"
                        src="test-file-stub"
                      />
                    </div>
                    <div
                      class="css-l273ov"
                    >
                      <p
                        class="css-iz8f7z"
                      >
                        MSP
                        -
                        4
                      </p>
                    </div>
                  </div>
                  <div
                    class="css-1e2fy0l"
                  />
                </div>
              </a>
              <a
                class="css-1u1ubl7"
                defaultcolor="#E36060"
                href="/easy-project/dashboard/2/1111"
              >
                <div
                  class="css-1gtjcrg"
                />
                <div
                  class="css-1lojfjy"
                >
                  <p
                    class="css-1l5tvrx"
                  >
                    Не загружаются задачи при клике
                  </p>
                </div>
                <div
                  class="css-1ox63rq"
                >
                  <div
                    class="css-1c3mbim"
                  >
                    <div
                      class="css-1e2fy0l"
                    >
                      <img
                        class="css-6jm447"
                        src="test-file-stub"
                      />
                    </div>
                    <div
                      class="css-l273ov"
                    >
                      <p
                        class="css-iz8f7z"
                      >
                        MSP
                        -
                        3
                      </p>
                    </div>
                  </div>
                  <div
                    class="css-1e2fy0l"
                  >
                    <div
                      class="css-11um8xy"
                    />
                  </div>
                </div>
              </a>
            </div>
          </div>
          <div
            class="css-1ni1msf"
          >
            <div>
              <p
                class="css-1xdtxed"
              >
                TESTING (1)
              </p>
              <a
                class="css-1u1ubl7"
                defaultcolor="#E36060"
                href="/easy-project/dashboard/2/111111"
              >
                <div
                  class="css-1gtjcrg"
                />
                <div
                  class="css-1lojfjy"
                >
                  <p
                    class="css-1l5tvrx"
                  >
                    Неправильная ссылка на дашбор списка задач
                  </p>
                </div>
                <div
                  class="css-1ox63rq"
                >
                  <div
                    class="css-1c3mbim"
                  >
                    <div
                      class="css-1e2fy0l"
                    >
                      <img
                        class="css-6jm447"
                        src="test-file-stub"
                      />
                    </div>
                    <div
                      class="css-l273ov"
                    >
                      <p
                        class="css-iz8f7z"
                      >
                        MSP
                        -
                        6
                      </p>
                    </div>
                  </div>
                  <div
                    class="css-1e2fy0l"
                  >
                    <div
                      class="css-11um8xy"
                    />
                  </div>
                </div>
              </a>
            </div>
          </div>
          <div
            class="css-1ni1msf"
          >
            <div>
              <p
                class="css-1xdtxed"
              >
                DONE (1)
              </p>
              <a
                class="css-1u1ubl7"
                defaultcolor="#93E586"
                href="/easy-project/dashboard/2/11111"
              >
                <div
                  class="css-6sdg1k"
                />
                <div
                  class="css-1lojfjy"
                >
                  <p
                    class="css-1l5tvrx"
                  >
                    Hello, world!
                  </p>
                </div>
                <div
                  class="css-1ox63rq"
                >
                  <div
                    class="css-1c3mbim"
                  >
                    <div
                      class="css-1e2fy0l"
                    >
                      <img
                        class="css-6jm447"
                        src="test-file-stub"
                      />
                    </div>
                    <div
                      class="css-l273ov"
                    >
                      <p
                        class="css-iz8f7z"
                      >
                        MSP
                        -
                        5
                      </p>
                    </div>
                  </div>
                  <div
                    class="css-1e2fy0l"
                  >
                    <div
                      class="css-11um8xy"
                    />
                  </div>
                </div>
              </a>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
